manager/allocator/network.go

```diff
@@ -758,8 +847,48 @@ func (a *Allocator) doTaskAlloc(ctx context.Context, ev events.Event) {
 	nc.pendingTasks[t.ID] = t
 }
 
-func (a *Allocator) allocateNode(ctx context.Context, node *api.Node) error {
-	return a.netCtx.nwkAllocator.AllocateNode(node)
+func (a *Allocator) allocateNode(ctx context.Context, node *api.Node, existingAddressesOnly bool, networks []*api.Network) bool {
+	var allocated bool
+
+	nc := a.netCtx
+
+	for _, network := range networks {
+
+		var lbAttachment *api.NetworkAttachment
+		for _, na := range node.LbAttachments {
+			if na.Network.ID == network.ID {
+				lbAttachment = na
+				break
+			}
+		}
+
+		if lbAttachment != nil {
+			if nc.nwkAllocator.IsLBAttachmentAllocated(node, lbAttachment) {
+				continue
+			}
+		}
+
+		if lbAttachment == nil {
+			lbAttachment = &api.NetworkAttachment{}
+			node.LbAttachments = append(node.LbAttachments, lbAttachment)
+		}
+
+		if existingAddressesOnly && len(lbAttachment.Addresses) == 0 {
+			continue
+		}
+
+		lbAttachment.Network = network.Copy()
+		if err := a.netCtx.nwkAllocator.AllocateLBAttachment(node, lbAttachment); err != nil {
+			log.G(ctx).WithError(err).Errorf("Failed to allocate network resources for node %s", node.ID)
+			//TODO: Should we add a unallocatedNode and retry allocating resources like we do for network, tasks, services?
```

Comment:
@nishanttotla : Seems like we should have some retry logic here.  As far as I can tell, the original code does not have any retry logic if we fail to allocate network resources for the ingress network.

Reply:
@pradipd if retry logic does not exist, perhaps this PR doesn't need to add it. If I understand correctly, not having retry logic here will mean that if the LB endpoint attachment fails, then the endpoint does not work. Is there a way to allocate later in that case? What is the fix in case of failure?

Reply:
In the existing code, if we fail to allocate the endpoint for the ingress network, the node still becomes active, and we can schedule tasks on it.  But, the node, won't be part of an load balancers (ILB or ELB(routing mesh)).  This would be "bad" (I know that's arbitrary) for a customer, because they may not realize some node is not participating in the routing mesh.

My change does nothing to add this, other than leave a TODO.
If we were to address this, we should add retry logic similar to unallocatedTasks, unallocatedServices, and unallocatedNetworks.  See L317 when we retry unallocated items after a timeout or when something was deleted.

Also, I feel addressing this should be a separate issue and have a separate PR.

Reply:
@pradipd I agree. Let's keep that separate.

cc @mavenugo @vieux for confirmation on this. We should track this as a follow-up because the load balancing may not work properly without it.